lib/timeutils.c

```diff
@@ -147,7 +147,27 @@ static int parse_sec(const char *t, usec_t *usec)
 	return 0;
 }
 
-int parse_timestamp(const char *t, usec_t *usec)
+static int parse_subseconds(const char *t, usec_t *usec)
+{
+	usec_t ret = 0;
+	int factor = USEC_PER_SEC / 10;
+
+	if (*t != '.' && *t != ',')
+		return 0;
+
+	while (*(++t)) {
+		if (!isdigit(*t) || factor < 1)
+			return 0;
+
+		ret += ((usec_t) *t - '0') * factor;
+		factor /= 10;
+	}
+
+	*usec = ret;
+	return 1;
```

Comment:
Would it be possible to return 0 on success and <0 on error? ... as we usually use ;-)